samtools/htsjdk
# replacing ant build with gradle build

Major changes to htsjdk's build:
- updating to java 8
- adding a build.gradle
  - restructuring repo to use gradle default style:
    - src/java -> src/main/java
    - src/tests/java -> src/test/java
    - test-data -> src/test/resources
- removing ant's build.xml
- removing etc/ because it looks like it's for help formatting the test results nicely, and gradle does nice job already
- updating .travis.yml to use gradle
- adding gradlew and gradlew.bat wrapper files, these let users without gradle run ./gradlew instead
- removing intellij and eclipse project files
- deleting lib because we're pulling down our dependencies from maven
- removing intel deflator because it's comments say it's not needed in java 8
  - removing associated c and jni code
- update gitignore with gradle relevant directories

Things not done yet:
- remove sbt
I'd like to remove the sbt code entirely and replace with equivalent gradle code.  This will probably take some coordination with the @jacarey since he handles all the releases and I don't know the process exactly / have signing keys 
- update documentation 
I don't know how to update the github-io site, can that be done from within a pull request?
- building picard on the gradle version
We need to try this to make sure it will work.  Probably take some minor changes to the picard build to make sure jar paths are correct.
- probably other things...


[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/samtools/htsjdk/383)
